lib/Parser/Scan.h

```diff
@@ -367,10 +367,13 @@ class Scanner : public IScanner, public EncodingPolicy
     {
         return HeapNewNoThrow(Scanner, parser, phtbl, ptoken, perr, scriptContext);
     }
+
+CLANG_WNO_BEGIN("-Wall") // -Wno-non-virtual-dtor
```

Comment:
Just alternatives for suppressing the warning. Searching on the web shows options such as adding a public virtual dtor or a protected non virtual dtor. Either looks better than code suppressing warnings.